GitUI/CommandsDialogs/FormCommit.cs

```diff
@@ -896,6 +912,21 @@ protected override bool ExecuteCommand(int cmd)
             }
         }
 
+        public override IScriptOptionsProvider? GetScriptOptionsProvider(GitModuleControl source)
+        {
+            FileStatusList? fileStatusList = _currentFilesList;
+            for (Control control = source; control is not null && control != this; control = control.Parent)
+            {
+                if (control is FileStatusList controlFileStatusList)
+                {
+                    fileStatusList = controlFileStatusList;
+                    break;
+                }
+            }
+
```

Comment:
I doubt this is still necessary. Then the argument `source` becomes obsolete, too.

Reply:
It is. It is used when a script is activated by a hotkey. By looking for the nearest parent, if `Staged` or `Unstaged` are focused, the matching files are used in the script. While if the diff view or commit textbox are selected, it uses the last focused list, or `null` if no list was focused yet.

Reply:
With 11363, there is always an active FileStatusList. So there is no need for the argument `GitModuleControl source` any longer.

Reply:
There's always an active one, but that doesn't mean it's the same one that intercepted the hotkey. This basically removes the necessity to implement `IScriptOptionProvider` on each child. Event bubbling is a common pattern and, honestly, should have probably been used for the hotkey implementation itself.

Reply:
The current FileStatusList should be the focused one. The FileStatusList selection controls the diff. Hence in FormCommit, it is all about `_currentFilesList` regardless of which control actually has the focus.

Hotkeys *are* handled by the focused control or *do* bubble up to the parents, then to the Form and some Forms even forward them downwards to active controls.

Reply:
`ExecuteScriptCommand` is implemented and called within `GitModuleControl.ExecuteCommand`. This, in effect, means that every control, without exception, handles every script hotkey. If the source control did not matter, this could have been handled in the form. Although, in such a case, I would have needed to trap it in order for `IScriptOptionsProvider` to have an effect. In effect, bubbling is not used in the hotkey implementation, at least for scripts. This is why in `GitModuleControl.ExecuteCommand`, I ended up adding a bubbling call to `GetScriptOptionsProvider` instead.

A side effect of this is that script hotkeys take precedence over form or parent hotkeys, despite taking effect after the current control's hotkeys.

An alternative implementation might have been for `ExecuteScriptCommand` to immediately return `false` for any control that isn't an `IScriptOptionsProvider` (or returns `null` from `GetScriptOptionsProvider`). This would make it automatically bubble up to the nearest `IScriptOptionsProvider`, or be handled by the form.

I think the reason this isn't done is because `ExecuteCommand` is also directly used for script context menus (passed as `scriptInvoker`), and bubbling does not occur there. Same for `ShowInUserMenuBar`

Reply:
> An alternative implementation might have been for ExecuteScriptCommand to immediately return false for any control that isn't an IScriptOptionsProvider (or returns null from GetScriptOptionsProvider). This would make it automatically bubble up to the nearest IScriptOptionsProvider, or be handled by the form.

I would prefer this approach.

> I think the reason this isn't done is because ExecuteCommand is also directly used for script context menus (passed as scriptInvoker), and bubbling does not occur there. Same for ShowInUserMenuBar

I guess this was just added in a place where it worked - without thinking about future extensions.

Reply:
There's still a bit of ugliness, because C# does not have traits, so `GitExtensionsControl` and `GitExtensionsFormBase` have a lot of implementation overlap. But accepted